File: src/main/java/chat/teco/tecochat/chat/application/chat/dto/ChatSocketContext.java
package chat.teco.tecochat.chat.application.chat.dto;

import chat.teco.tecochat.domain.member.Member;
import org.springframework.web.socket.WebSocketSession;

public class ChatSocketContext {

    private final WebSocketSession session;
    private final Member member;
    private final StringBuilder stringBuilder;
    private final Long chatId;
    private String question;

    public ChatSocketContext(WebSocketSession session, Member member, Long chatId) {
        this.session = session;
        this.member = member;
        this.chatId = chatId;
        this.stringBuilder = new StringBuilder();
    }

    public String answer() {
        return stringBuilder.toString();
    }

    public Member member() {
        return member;
    }

    public Long chatId() {
        return chatId;
    }

    public String question() {
        return question;
    }

    public WebSocketSession session() {
        return session;
    }

    public StringBuilder stringBuilder() {
        return stringBuilder;
    }

    public void setQuestion(String question) {
        this.question = question;
    }
}

File: src/main/kotlin/chat/teco/tecochat/infra/gpt/ChatSocketContext.kt
package chat.teco.tecochat.infra.gpt

import chat.teco.tecochat.domain.member.Member
import org.springframework.web.socket.TextMessage
import org.springframework.web.socket.WebSocketSession
import java.io.IOException

class ChatSocketContext(
    val session: WebSocketSession,
    val member: Member,
    val chatId: Long?,
) {
    private val answers: MutableList<String> = mutableListOf()
    private var question: String = ""

    fun addAnswer(answer: String) {
        answers.add(answer)
    }

    fun changeQuestion(question: String) {
        this.question = question
    }

    fun sendMessage(message: String) {
        try {
            session.sendMessage(TextMessage(message))
        } catch (e: IOException) {
            throw IllegalStateException(e)
        }
    }

    fun close() {
        try {
            session.close()
        } catch (e: IOException) {
            throw IllegalStateException(e)
        }
    }

    fun getAnswer() = answers.joinToString("")

    fun getCurrentQuestion() = question
}
